[ty] Add error context for TypedDict assignments (#24790)
astral-sh/ruff: crates/ty_python_semantic/resources/mdtest/diagnostics/error_context.md

```toml
[environment]

> python-version = "3.13"

Author: For `typing.ReadOnly`

   |             |
   |             Declared type
   |

> info: field "name" on TypedDict `Person` has type `str` which is not assignable to type `bytes` expected by TypedDict `Other`

Reviewer: or maybe

```suggestion
info: Incompatible types for the `name` field
info: └── `str` is not assignable to `bytes`
```

?

Author: no, that would contradict the general structure. If `str` and `bytes` were more complex types, we would show a subtree, but we don't do that for basic types.

Reviewer: hmm, okay. it _just about_ fits onto a single line on a terminal screen if I have the terminal full-screen:

<img width="2298" height="382" alt="image" src="https://github.com/user-attachments/assets/5fb190f1-a34c-4928-bf01-3d5df22d6dd4" />

but not at all if I'm split-screening (but then, I'm sure that also applies for other diagnostic messages we show -- including, in this case, the summary message:

<img width="1314" height="448" alt="image" src="https://github.com/user-attachments/assets/b9f38375-b3fa-4aed-acba-cbf629d61be4" />
